File: api/src/main/java/com/transferschedule/api/models/Transacao.java
package com.transferschedule.api.models;

import java.math.BigDecimal;
import java.util.Date;

import com.transferschedule.api.enums.TIPOOPERACAO;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@Setter
@Getter
@Table(name = "transacao")
@Entity(name = "transacao")
public class Transacao extends BaseEntity {
    @Column(name = "cod_transacao")
    public long codTransacao;
    @Column(name = "num_val_transferencia")
    public BigDecimal numValTransferencia;
    @Column(name = "num_val_taxa_prevista")
    public BigDecimal numValTaxaPrevista;
    @Column(name = "dta_transacao")
    public Date dtaTransacao;

    public TIPOOPERACAO indTipoOperacao;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conta_origem_uuid") // Mapeamento para a chave estrangeira da conta de origem
    private Conta contaOrigem;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "conta_destino_uuid") // Mapeamento para a chave estrangeira da conta de destino
    private Conta contaDestino;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cliente_responsavel_transacao_uuid")
    public Cliente cliente;



    @Override
    public void isValid() {

    }
}

File: web/src/models/transacao.ts
export class Transacao {
  constructor(
    public uuid: string,
    public codTransacao: number,
    public numValTransferencia: number,
    public numValTaxaPrevista: number,
    public dtaTransacao: Date,
    public contaOrigemUuid: string,
    public contaDestinoUuid: string,
    public clienteResponsavelUuid: string,
    public indTipoOperacao?: TIPOOPERACAO,
    public dtaCreateAt?: Date,
    public dtaUpdateAt?: Date,
    public dtaDeleteAt?: Date,
  ) {}
}

export enum TIPOOPERACAO {
  A = 1,
  B = 2,
  C = 3,
  D = 4,
}
